alacritty_terminal/src/keyboard_motion.rs

```diff
@@ -0,0 +1,643 @@
+                }
+            },
+            KeyboardMotion::Down => {
+                if self.point.line >= lines - 1 {
+                    term.scroll_display(Scroll::Lines(-1));
+                } else {
+                    self.point.line += 1;
+                }
+            },
+            KeyboardMotion::Left => {
+                let mut buffer_point = term.visible_to_buffer(self.point);
+                buffer_point = expand_wide(term, buffer_point, true);
+                buffer_point.col = Column(buffer_point.col.saturating_sub(1));
+                self.point = term.buffer_to_visible(buffer_point).unwrap_or_default().into();
+            },
+            KeyboardMotion::Right => {
+                let mut buffer_point = term.visible_to_buffer(self.point);
+                buffer_point = expand_wide(term, buffer_point, false);
+                buffer_point.col = min(buffer_point.col + 1, cols - 1);
+                self.point = term.buffer_to_visible(buffer_point).unwrap_or_default().into();
+            },
+            KeyboardMotion::Start => self.point.col = Column(0),
+            KeyboardMotion::End => self.point.col = cols - 1,
+            KeyboardMotion::High => self.point = Point::new(Line(0), Column(0)),
+            KeyboardMotion::Middle => self.point = Point::new(Line((lines.0 - 1) / 2), Column(0)),
+            KeyboardMotion::Low => self.point = Point::new(lines - 1, Column(0)),
+            KeyboardMotion::SemanticLeft => {
+                self.point = Self::semantic_move(term, self.point, true, true)
+            },
+            KeyboardMotion::SemanticRight => {
+                self.point = Self::semantic_move(term, self.point, false, true)
+            },
+            KeyboardMotion::SemanticLeftEnd => {
+                self.point = Self::semantic_move(term, self.point, true, false)
+            },
+            KeyboardMotion::SemanticRightEnd => {
+                self.point = Self::semantic_move(term, self.point, false, false)
+            },
+            KeyboardMotion::WordLeft => self.point = Self::word_move(term, self.point, true, true),
+            KeyboardMotion::WordRight => {
+                self.point = Self::word_move(term, self.point, false, true)
+            },
+            KeyboardMotion::WordLeftEnd => {
+                self.point = Self::word_move(term, self.point, true, false)
+            },
+            KeyboardMotion::WordRightEnd => {
+                self.point = Self::word_move(term, self.point, false, false)
+            },
+            KeyboardMotion::Bracket => {
+                if let Some(p) = term.bracket_search(term.visible_to_buffer(self.point)) {
+                    // Scroll viewport if necessary
+                    scroll(term, p);
+
+                    self.point = term.buffer_to_visible(p).unwrap_or_default().into();
+                }
+            },
+        }
+
+        self
+    }
+
+    /// Move by semanticuation separated word, like w/b/e/ge in vi.
```

Comment:
>semanticuation

Is this even a word? I feel like it's not, since search engine isn't aware of it. maybe `semantically`?

Reply:
Looks like a regex replace messup. :smile: